### PR TITLE
Test Kitchen Integration

### PR DESCRIPTION
**Pull request by adamleff**
Supersedes #6.

Support for Test Kitchen integration/functional testing using new cisco-specific gems, and initial contribution of serverspec tests for current cookbook custom resources.

### CODE REVIEW

**Comment by glennmatthews** [comment]
:+1: from me (the Rubocop failures look pretty trivial to fix post-merge) - Rob, if you're ok with it too, go ahead and merge.

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+ - Checkout the complete cookbook, including the this "test" folder
+ - Create a bundle environment to run the tests from using ```chef exec bundle``` from this main cookbook directory (the Gemfile there will be used to create the bundle)
+ - Modify .kitchen.yml to point at switch under test's IP address and BASH login credentials
+ - Modify .kitchen.yml to use appropriate VRF to connect to Internet via switch under test (if not management VRF)
+
+## First run procedure
+
+ - Converge test cookbook to switch and verify using serverspec using ```chef exec bundle exec kitchen converge``` from the cookbook directory
+ - At this point, all tests should pass. You can cleanup by running ```chef exec bundle exec kitchen shutdown``` from the cookbook directory and removing /tmp/verifier directory on switch under test
```
Comment (robert-w-gries) [inline test/README.md]: This command didn't work for me.  Is it going to be implemented?

```
~/forks/chef-partners/cisco-network-chef-cookbook$ chef exec bundle exec kitchen shutdown
Could not find command "shutdown".
```

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Testing using Test kitchen
+
+## Requirements
+
+ - ChefDK installed on local workstation
+ - Switch under test available via SSH
+ - Switch under test needs valid /etc/resolv.conf
+ - Switch under test needs Internet access (for rubygems) via some VRF
```
Comment (robert-w-gries) [inline test/README.md:8]: Can you add the following as a requirement?

`- Switch under test needs a user configured with bash login (Ex: username kitchen shelltype bash)`

File: test/README.md (hunk 1 above)
Comment (adamleff) [inline test/README.md]: Actual command is `kitchen destroy` not "shutdown" - I will fix that.

File: test/README.md (hunk 2 above)
Comment (adamleff) [inline test/README.md:8]: You bet!

**Pull request merged**